baseapp/grpcrouter.go

```diff
@@ -81,18 +84,23 @@ func (qrt *GRPCQueryRouter) RegisterService(sd *grpc.ServiceDesc, handler interf
 			// call the method handler from the service description with the handler object,
 			// a wrapped sdk.Context with proto-unmarshaled data from the ABCI request data
 			res, err := methodHandler(handler, sdk.WrapSDKContext(ctx), func(i interface{}) error {
-				err := qrt.cdc.Unmarshal(req.Data, i)
-				if err != nil {
-					return err
+				if msgv2, ok := i.(proto.Message); ok {
```

Comment:
rather than adding the logic here, I would add it to the GRPCCodec here: https://github.com/cosmos/cosmos-sdk/blob/master/codec/proto_codec.go#L269

Reply:
We can but it wouldn't be full support. @technicallyty has a PR with more full support but it's significantly more code changes that we can't backport really I think

Reply:
Also given how things have evolved I'm not sure there's much value in making Codec proto v2 compatible. For ORM, we don't really need a codec thing but rather simply a type resolver. The other boundaries are query and msg routers and those could maybe be the touch points where we check the proto api. 

The main reason for using codec would maybe be if you wanted to use a pulsar message inside a gogo any. But since we won't be defining interface methods on pulsar types, not sure that's realistic. 

Other than that I don't see a reason for pulsar code to interact with a codec explicitly. All of the unknown field rejection, etc should happen at a level before unmarshaling btw so also doesn't need the existing codec - just a registry

Reply:
Yes and I understand this, but what I linked there is the `GRPCCodec` not the `Codec`, we use it to properly support gogo, proto and proto2. Instead of adding this logic outside we could concentrate it into `GRPCCodec`, so other components (such as client.Context) can make use of it, without the need to replicate the logic.